### ISSUE TITLE
Merge to empty array fails

### ISSUE THREAD

**Issue by hobti01** [mikefarah/yq#670]
**Describe the bug**
I'd like to merge multiple files.  Previous files do not necessarily have all keys of later files.

Version of yq: 4.4.0
Operating system: mac
Installed via: homebrew

**Input Yaml**
Concise yaml document(s) (as simple as possible to show the bug, please keep it to 10 lines or less)
data1.yml:
```yaml
# first file cannot be empty or only comments
no: op
```

data2.yml:
```yaml
arr:
  - item1
```

**Command**
The command you ran:
```
yq eval-all 'select(fileIndex == 0) *+ select(fileIndex == 1)' data1.yml data2.yml
```

**Actual behavior**

```
Error: !!seq ([arr]) cannot be added to a !!null
```

**Expected behavior**

```yaml
no: op
arr:
  - item1
```

**Additional context**
Ideally I'd like to merge a list of files instead of looping in bash, not requiring that the files have content.  Something like
```
yq eval-all "select(fileIndex == *) *+" /dir/*.yaml
# or merge without append:
yq eval-all "select(fileIndex == *) *" /dir/*.yaml
```

thanks for your effort!

**Comment by mikefarah** [comment]
Got it thanks.

Not sure how that would work in yq, have to think about that

**Comment by mikefarah** [comment]
Ah I think it's like `jq`' reduce functionality: https://stedolan.github.io/jq/manual/#Reduce

That will be tricky to implement...very cool though

**Comment by mikefarah** [comment]
Fixed in 4.4.1 (nb not the merge list of files with no loops - feel free to raise a separate card)
